src/sync.cpp

```diff
@@ -701,21 +701,21 @@ Sync::Sync(MegaClient* cclient, SyncConfig config, const char* cdebris,
     if (macOSmajorVersion() >= 19) //macOS catalina+
     {
         LOG_debug << "macOS 10.15+ filesystem detected. Checking fseventspath.";
-        string supercrootpath = "/System/Volumes/Data" + crootpath;
+        string supercrootpath = "/System/Volumes/Data" + *crootpath.editStringDirect();
 
         int fd = open(supercrootpath.c_str(), O_RDONLY);
         if (fd == -1)
         {
             LOG_debug << "Unable to open path using fseventspath.";
-            mFsEventsPath = crootpath;
```

Comment:
Looks good, the only thing I would say is that it might be convenient to make mFsEventsPath a LocalPath instead of a string, but this is fine for now too.  Thanks for fixing this up.